middleware/index.ts

```diff
@@ -130,19 +130,22 @@ function createSdkContext(reqContext: any, accessToken?: string, functionInvocat
         );
         unitOfWork = new UnitOfWork(config, logger);
         forceApi = new ForceApi(config, logger);
-
-        if (functionInvocationId) {
-            fxInvocation = new FunctionInvocationRequest(functionInvocationId, logger, forceApi);
-        }
     }
 
-    return new SdkContext(apiVersion,
-                          userCtx, 
-                          logger, 
-                          reqContext.payloadVersion,
-                          forceApi, 
-                          unitOfWork,
-                          fxInvocation);
+    let initedSdkContext: SdkContext = new SdkContext(apiVersion,
+                                                      userCtx,
+                                                      logger,
+                                                      reqContext.payloadVersion,
+                                                      forceApi,
+                                                      unitOfWork);
+
+    // if functionInvocationId is set,
+    // dynamically set "fxInvocation" object as common code for internal hello and pdf function to share
```

Comment:
Let's leave out that `FunctionInvoctionRequest` is internal and who uses it.  Ie, in comment, remove ref to internal and specific functions that use `FunctionInvoctionRequest`.